docs/madr/decisions/046-policy-on-namespace.md

```diff
@@ -0,0 +1,456 @@
+the namespaced MeshTimeout is considered more specific.
+That's why the final request timeout value should be 3s.
+
+**4. Both MeshTimeout and MeshHTTPRoute in "frontend-ns"**
+
+This scenario is likely to be the most common way users create policies.
+The use case for this is:
+
+**As** a Frontend Service Owner<br>
+**I want** all outgoing requests from fronted (to backend) to have headers starting with `xxx-fronted-` removed<br>
+**so that** requests are smaller and backend doesn't receive unnecessary data.
+
+**As** a Frontend Service Owner<br>
+**I want** all requests routed to `backend` to have 3s timeout<br>
+**so that** my service meets SLO and responds withing 3s.
+
+This can be achieved by applying 2 policies:
+
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: MeshHTTPRoute
+metadata:
+  name: backend-route
+  namespace: frontend-ns
+spec:
+  targetRef:
+    kind: MeshService
+    name: frontend
+  to:
+    - targetRef:
+        kind: MeshService
+        name: backend
+      rules:
+        - matches:
+            - path:
+                type: PathPrefix
+                value: /
+          default:
+            filters:
+              - type: RequestHeaderModifier
+                requestHeaderModifier:
+                  remove:
+                    - xxx-frontend-version
+                    - xxx-frontend-data
+                    - xxx-frontend-content
+            backendRefs:
+              - kind: MeshService
+                name: backend
+---
+apiVersion: kuma.io/v1alpha1
+kind: MeshTimeout
+metadata:
+  name: frontend-to-backend
+  namespace: frontend-ns
+spec:
+  targetRef:
+    kind: MeshHTTPRoute
+    name: backend-route
+    namespace: frontend-ns # can be omitted as it's a default value
+  to:
+    - targetRef:
+        kind: Mesh
+      default:
+        http:
+          requestTimeout: 3s
+```
+
+**5. MeshTimeout in "frontend-ns", MeshHTTPRoute in "backend-ns"**
+
+MeshHTTPRoute in `backend-ns` produces envoy routes in DPPs in `backend-ns`.
+Ability to configure timeouts on these routes from the `frontend-ns` breaks the namespace isolation and seems
+unnecessary.
+
+**In the end,** only 3 scenarios have meaningful use cases and could be described by 2 principles:
+
+* a policy in a custom namespace can reference all other policies regardless their namespace
+* a policy in the system namespace can reference only policies in the system namespace
+
+### MeshService reference
+
+Since we didn't make any decisions on how to target MeshService in policies we will leave this out of now. We will
+revisit this while discussing targeting MeshService in policies.
+
+### MeshGateway reference
+
+`MeshGateway` at this moment is not namespace-scoped. It can change in the future with
+the [MeshBuiltinGateway resource](https://github.com/kumahq/kuma/issues/10014).
+
+We've decided that we won't allow targeting `MeshGateway` in namspace scoped polcies except for the
```

Comment:
it's fine by me, but I'm just curious. Why only MeshHTTPRoute and MeshTCPRoute. How are those different than policies?

Reply:
@Automaat added some explanation here https://github.com/kumahq/kuma/pull/10148/files#diff-cb6c27d1ae4efdd51ae3730849991f19c4016f47fb91292fa349d9c0e0ea8a95R318-R320, do it make sense?